projects/common/src/time/time-duration.ts

```diff
@@ -16,20 +16,35 @@ export class TimeDuration {
     return this.toMillis() / this.unitInMillis(unit);
   }
 
-  public toMultiUnitString(smallestUnit: ConvertibleTimeUnit = TimeUnit.Second, displayZero: boolean = true): string {
+  public toMultiUnitString(
+    smallestUnit: ConvertibleTimeUnit = TimeUnit.Second,
+    displayZero: boolean = true,
+    unitStringType: UnitStringType = UnitStringType.Short
+  ): string {
     const mostSignificantPortion = this.getMostSignificantUnitOnly();
     const remainingMillis = this.millis - mostSignificantPortion.toMillis();
     if (mostSignificantPortion.getAmountForUnit(smallestUnit) < 1) {
-      return displayZero ? new TimeDuration(0, smallestUnit).toString() : '';
+      return displayZero
+        ? unitStringType === UnitStringType.Long
+          ? new TimeDuration(0, smallestUnit).toLongString()
```

Comment:
nit: a private method that accepts the unitStringType (i.e `private toFormattedString(type: UnitStringType): string`) would allow removing all of the ternaries and particularly the nested one.

Reply:
did a bit of clean up here.